code/game/objects/structures/displaycase.dm

```diff
@@ -3,27 +3,40 @@
 	icon = 'icons/obj/structures/displaycase.dmi'
 	icon_state = "glassbox"
 	desc = "A display case for prized possessions. It taunts you to kick it."
-	density = 1
-	anchored = 1
-	unacidable = 1//Dissolving the case would also delete the gun.
+	density = TRUE
+	anchored = TRUE
+	unacidable = TRUE //Dissolving the case would also delete the gun.
 	alpha = 150
 	maxhealth = 100
 	hitsound = 'sound/effects/Glasshit.ogg'
-	var/destroyed = 0
+	req_access = null
+
+	var/destroyed = FALSE
+	var/locked = TRUE
 
 /obj/structure/displaycase/Initialize()
 	. = ..()
 	var/turf/T = get_turf(src)
 	for(var/atom/movable/AM in T)
 		if(AM.simulated && !AM.anchored)
 			AM.forceMove(src)
+	
+	if(!req_access)
+		var/area/A = get_area(src)
+		if(!istype(A) || !islist(A.req_access))
+			return
+		req_access = A.req_access.Copy()
+	
 	update_icon()
```

Comment:
Might need to move this before the req_access block, or change the block to not return.